### DISCUSSION TITLE
Can metrics be remapped to send multiple metrics?

### DISCUSSION THREAD

**Discussion by alex-rowe** [vectordotdev/vector#18642] [Q&A] [answered]
In the remap transform there is the option to split the log and send multiple by overriding the top level `.` to send an array instead of an object, does this also exist for metrics?

[/transforms/remap/#emitting-multiple-log-events](https://vector.dev/docs/reference/configuration/transforms/remap/#emitting-multiple-log-events)

For example:
```
[sources.internal_metrics]
type = "internal_metrics"

[transforms.filter_metrics]
type = "remap"
inputs = ["internal_metrics"]
source = '''
  # code removed to filter on just the metrics we want
  metric_one = .
  metric_two = .
  metric_one.tags.first_tag = "string"
  metric_two.tags.second_tag = "string"
  . = [metric_one, metric_two]
```

The use case for this would be for sending metrics with two different single dimensions, rather than multiple dimensions.

An option may be two create three transforms, so `filter_metrics` to just remove metrics we don't want, then use it as the `input` for two other `transforms` where they set their tag. Those two `transforms` are then `inputs` to the `sink`.

**Comment by jszwedko** [comment, marked as answer]
Hi @alex-rowe !

I don't think `remap` can natively do that at the moment. Your work-around of using multiple transforms seems like it would work.

**Reply by alex-rowe** [reply]
Thanks, I tested the multiple transforms and that works well, one for a filter condition, then two for remapping to different value pairs.
